app/js/components/names.jsx

```diff
@@ -1,265 +1,251 @@
-                    <label className="control-label col-sm-2"> Created By: </label>
-                    <div className="col-sm-5">
-                        <input  className="form-control"
-                                name="createdby"
-                                type="text"
-                                size="50"
-                                value= {this.state.creator +'  ' + this.state.dateCreated}
-                                disabled />
-                    </div>
-                </div>
-                {this.state.display == "edit" &&
-                <div className="form-group">
-                    <div className="col-sm-2">
-                        <button type="submit"
-                             name="update"
-                             onClick={this.handleSave}
-                             className="btn btn-success form-control">
-                             Save</button>
-                    </div>
-                    <div className="col-sm-2">
-                        <button type="button"
-                             name="cancel"
-                             onClick={this.handleCancel}
-                             className="btn btn-default form-control cancelBtn">
-                             Cancel</button>
-                    </div>
-               </div>
-            }
-
-            </form>
-            </div>
-        );
-    };
+    if(!this.state.familyName) {
+      this.setState((prevState) => ({
+        familyName: this.state.familyName
+      }));
+      this.setState({ errorFamilyName: 'Family Name can not be null' })
+    }
+  }
+
+  handleCancel() {
+    this.fetch();
+    const given = this.state.res.preferredName.givenName;
+    const middle = this.state.res.preferredName.middleName;
+    const family = this.state.res.preferredName.familyName;
+    this.setState((prevState) => ({
+      display: 'view',
+      givenName: given === null ? '' : prevState.givenName,
+      middleName: middle === null ? '' : prevState.middleName,
+      familyName: family === null ? '' : prevState.familyName,
+      voided: prevState.res.preferredName.voided,
+      preferred: prevState.preferred,
+      creator: prevState.res.auditInfo.creator.display,
+      dateCreated: prevState.res.auditInfo.dateCreated,
+      errorGivenName: '',
+      errorFamilyName: ''
+    }));
+  };
+
+  render() {
+    let givenErrorClass = '';
+    let familyErrorClass = '';
+    if(this.state.errorGivenName && this.state.display === 'edit') {
+      givenErrorClass = 'has-error'
+    }
+    if(this.state.errorFamilyName && this.state.display === 'edit') {
+      familyErrorClass = 'has-error'
+    }
+    return(
+      <div>
+        <form className='form-horizontal'>
+        {this.state.display === 'view' &&
+          <div className='form-group'>
+              <div className='col-sm-2'>
+                  <input className='btn btn-success form-control'
+                         name='edit'
```

Comment:
I think you may consider the right indentation here

Reply:
Done